internal/provider/waypoint/resource_waypoint_action.go

```diff
@@ -0,0 +1,629 @@
+		methodString = types.StringValue("POST")
+	case waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPUT:
+		methodString = types.StringValue("PUT")
+	case waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPATCH:
+		methodString = types.StringValue("PATCH")
+	case waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodDELETE:
+		methodString = types.StringValue("DELETE")
+	default:
+		return methodString, fmt.Errorf("unknown method")
+	}
+	return methodString, nil
+}
+
+func convertMethodToEnumType(method string) (waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethod, error) {
+	var methodEnum waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethod
+	switch method {
+	case "GET":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodGET
+	case "POST":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPOST
+	case "PUT":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPUT
+	case "PATCH":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPATCH
+	case "DELETE":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodDELETE
+	default:
+		return methodEnum, fmt.Errorf("unknown method")
+	}
+	return methodEnum, nil
+}
+
+func (r *ActionResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_waypoint_action"
+}
+
+func (r *ActionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "The Waypoint Action resource managed the lifecycle of an Action.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the Action.",
+				Computed:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Action.",
+				Required:    true,
+			},
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the HCP project where the Action is located.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"organization_id": schema.StringAttribute{
+				Description: "The ID of the HCP organization where the Action is located.",
+				Computed:    true,
+			},
+			// An Action description must be fewer than 125 characters if set.
+			"description": schema.StringAttribute{
+				Description: "A description of the Action.",
+				Optional:    true,
+			},
+			"namespace_id": schema.StringAttribute{
+				Description: "Internal Namespace ID",
+				Computed:    true,
+			},
+			// Action URL is required if the action is custom
+			"action_url": schema.StringAttribute{
+				Description: "The URL to trigger an action on. Only used in custom mode",
+				Optional:    true,
+			},
```

Comment:
Part of me is thinking maybe we don't allow users to set this at all, does it need to be present in the schema because it exists under the Action config, or is it just a field that's user-set-able? If it's the second one, maybe we remove it from the provider.

Reply:
This could certainly be removed, it is present in the provider because the action config endpoint accepts it and I wasn't sure if it was redundant with the URL within the `request`. I will get rid of it in that case, the provider will function fine without it.

Reply:
Yep, originally before the `Request` struct was present in actions, we just had `action_url`. Now it's kind of turned into a shortcut, where if you set it with no Request struct, we auto-create a simple GET request on that url. But maybe for the TF provider that shortcut might be more confusing 🤔